src/military/military.cpp

```diff
@@ -7267,4 +7349,42 @@ bool pop_eligible_for_mobilization(sys::state& state, dcon::pop_id p) {
 		&& pop.get_poptype().get_strata() == uint8_t(culture::pop_strata::poor);
 }
 
+void disband_regiment_w_pop_death(sys::state& state, dcon::regiment_id reg_id) {
+	auto base_pop = state.world.regiment_get_pop_from_regiment_source(reg_id);
+	demographics::reduce_pop_size_safe(state, base_pop, int32_t(state.world.regiment_get_strength(reg_id) * state.defines.pop_size_per_regiment * state.defines.soldier_to_pop_damage));
+	state.world.delete_regiment(reg_id);
+}
+
+void disband_regiment_w_pop_transfer(sys::state& state, dcon::regiment_id reg_id) {
```

Comment:
Maybe I wasn't clear earlier: while this is an interesting idea, I don't want to add it as is at this point. I'm definitely open to adding new factors for how pops move around, but nothing as micro intensive as this.